modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -392,28 +399,95 @@ private JSONArray _getFiltersJSONArray(
 				Map<String, Object> properties =
 					fdsFilterObjectEntry.getProperties();
 
-				if (!Objects.equals(
-						MapUtil.getString(properties, "type"), "date")) {
+				String type = MapUtil.getString(properties, "type");
 
-					return null;
+				if (Objects.equals(type, "date")) {
+					return JSONUtil.put(
+						"entityFieldType", type
+					).put(
```

Comment:
Oh boy, we have a whole mess with `type`:
- There is a `type` that comes from OpenAPI, that can be `string`, `object`, `array` etc.
- There is a `format` that comes from OpenAPI, that can be `date` or `date-time` etc. This is what you are reading here, as we are [setting it that way in frontend](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx#L181).
- `entityFieldType` comes from [DXP backend FDS API](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/constants/FDSEntityFieldTypes.java). It can be `collection`, `date` or `string`. It is an arbitrary set of type constants. This is manually set, and determines [format of odata string](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/SelectionFilter.js#L61-L77).

There is couple of things that don't look right in the code here:
- What if `format` (disguised as `type`) is `data-time`?
-  `format` (disguised as `type`) is not the same as `entityFieldType`.
- I guess we're using `type` check as a hack to see if object entry is a date filter. Similar to `listTypeDefinitionId` check for dynamic filter. Is there a more direct, more readable way to test? Would it maybe worth doing separate handling for two `fdsFilterObjectEntries.addAll`s? I don't have a strong opinion here, might be good as it is.

IMO, what should we do:
1. Return both `format` and `type` from frontend.
2. Build `entityFieldType` from these two values:
    - `type` == `string` + `format` == `date` or `date-time` -->  `entityFieldType` = `date`
    - `type` == `string` + no `format` -->  `entityFieldType` = `string`
    - `type` == `array` --> `entityFieldType` = `collection`.
    - Ignore `type` == `object`, for now
3. Plan to place explicit `entityFieldType` in DSM UI. Maybe we can refactor `entityFieldType` to match OpenAPI, but I'm not sure. The point of this prop is to be a manual override of odata format, compensating that some endpoint was oddly implemented.

Reply:
Yeah, I noticed these inconsistencies when developing this PR though I didn't know exactly what was going on. But it was clear that we were mixing concepts from the debugging sessions and the values I got.

In fact, the `entityFieldType` value is not used in the frontend to decide the filter to use. It's the `type` value (see some lines below `entityFieldType`) what drives the type of filter to use. So, even if I'm adding an `entityFieldType` in L443 I don't think it is used for anything in the frontend side.

In any case, I'll hardcode the `entityFieldType` as you suggest for this PR, then it can be changed as needed in the future.